### DISCUSSION TITLE
Failed to Fetch Content - /api/framework/v1/read/NCF

### DISCUSSION THREAD

**Discussion by swayangjit** [Sunbird-Knowlg/Community#124] [Bugs]
The content load is failing. Please help in resolving the issue. Below are the details:

**URL**: https://<domain_name>/api/framework/v1/read/NCF?categories=board,gradeLevel,medium,class,subject

**Response**:
{
"id": "api.framework.read",
"ver": "1.0",
"ts": "2023-04-13T18:33:36.971Z",
"params": {
"resmsgid": "b42ce9b0-da29-11ed-9a8e-d53b9bbefcf9",
"msgid": null,
"status": "failed",
"err": null,
"errmsg": null
},
"responseCode": "SERVER_ERROR",
"result": {}
}

**Logs**: - [ERROR] api - {"id":"07530b15-2c52-0829-5ab0-211e71e15592","msg":"Getting error from ekstep while fetching framework by id ","additionalInfo":{"frameworkId":"NCF","queryString":""},"err":{"err":null,"responseCode":"SERVER_ERROR"}

**Reference**: https://github.com/orgs/Sunbird-Ed/discussions/461

**Comment by ruchinksinha** [comment]
@swayangjit
During the execution of the PostInstallScript, the job had failed multiple time and I tried to resolve all the issues. finally the job was successful. Now since the job was executed multiple time, its showing my as below in the application.

![image](https://user-images.githubusercontent.com/10114264/233537229-80a2766a-b835-4dd0-8e6f-939689ae9470.png)

I tried to retire all the existing frameworks using the retire API but its still showing multiple. So I wanted to cleanup all the framework and create fresh one. But now I am unable to create new framework because of the below error.

Request:
{
"request": {
"framework": {
"name": "NCF Demo",
"code": "ncfdemo",
"description": "NCF Demo Framework"
}
}
}

Response:
{
"id": "api.framework.create",
"ver": "1.0",
"ts": "2023-04-21T03:52:59.272Z",
"params": {
"resmsgid": "01c1e080-dff8-11ed-af80-afa5f3a40bf4",
"msgid": null,
"status": "failed",
"err": null,
"errmsg": null
},
"responseCode": "SERVER_ERROR",
"result": {}
}

Error in the knowledgemw corresponding to this API call is below:
[ERROR] api - {"id":"01c14440-dff8-11ed-a984-a754a0eac4d5","msg":"Getting error from ekstep while framework creation","additionalInfo":{"ekStepReqData":{"request":{"framework":{"name":"NCF Demo","code":"ncfdemo","description":"NCF Demo Framework"}}}},"err":{"err":null,"responseCode":"SERVER_ERROR"}}

**Reply by swayangjit** [reply]
@pallakartheekreddy Can you please look into this ?

**Reply by pallakartheekreddy** [reply]
@ruchinksinha @swayangjit please check the below discussion thread. We have provided steps to delete existing frameworks and create new ones
https://github.com/orgs/Sunbird-Knowlg/discussions/125

**Reply by pallakartheekreddy** [reply]
@swayangjit As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
